go-controller/pkg/ovn/obj_retry.go

```diff
@@ -126,18 +189,1067 @@ func (r *retryObjs) getObjRetryEntry(key string) *retryObjEntry {
+		objectsToRetry.labelSelectorForFilteredHandler, // filter out objects not matching these labels
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				oc.recordAddEvent(objectsToRetry.oType, obj)
+
+				key, err := getResourceKey(objectsToRetry.oType, obj)
+				if err != nil {
+					klog.Errorf("Upon add event: %v", err)
+					return
+				}
+				klog.Infof("Add event received for resource %v, key=%s", objectsToRetry.oType, key)
+
+				objectsToRetry.initRetryObjWithAdd(obj, key)
+				objectsToRetry.skipRetryObj(key)
+
+				// This only applies to pod watchers (pods + dynamic network policy handlers watching pods):
+				// if ovnkube-master is restarted, it will gets all the add events with completed pods
+				if oc.isObjectInTerminalState(objectsToRetry.oType, obj) {
+					oc.processObjectInTerminalState(objectsToRetry, obj, key, "add")
+					return
+				}
+
+				// If there is a delete entry with the same key, we got an add event for an object
+				// with the same name as a previous object that failed deletion.
+				// Destroy the old object before we add the new one.
+				if retryEntry := objectsToRetry.getObjRetryEntry(key); retryEntry != nil && retryEntry.oldObj != nil {
+					klog.Infof("Detected stale object during new object"+
+						" add of type %v with the same key: %s",
+						objectsToRetry.oType, key)
+					internalCacheEntry := oc.getInternalCacheEntry(objectsToRetry.oType, obj)
+					if err := oc.deleteResource(objectsToRetry, obj, internalCacheEntry); err != nil {
+						klog.Errorf("Failed to delete old object %s of type %v,"+
+							" during add event: %v", key, objectsToRetry.oType, err)
+						oc.recordErrorEvent(objectsToRetry.oType, obj, err)
+						objectsToRetry.unSkipRetryObj(key)
+						return
+					}
+					objectsToRetry.removeDeleteFromRetryObj(key)
+				}
+				start := time.Now()
+				if err := oc.addResource(objectsToRetry, obj, false); err != nil {
+					klog.Errorf("Failed to create %v object %s, error: %v",
+						objectsToRetry.oType, key, err)
+					oc.recordErrorEvent(objectsToRetry.oType, obj, err)
+					objectsToRetry.unSkipRetryObj(key)
+					return
+				}
+				klog.Infof("Creating %v %s took: %v", objectsToRetry.oType, key, time.Since(start))
+				objectsToRetry.deleteRetryObj(key, true)
+			},
+
+			UpdateFunc: func(old, newer interface{}) {
+				// skip the whole update if old and newer are equal
+				areEqual, err := areResourcesEqual(objectsToRetry.oType, old, newer)
+				if err != nil {
+					klog.Errorf("Could not compare old and newer resource objects of type %v: %v",
+						objectsToRetry.oType, err)
+					return
+				}
+				klog.V(5).Infof("Update event received for resource %s, old object is equal to new: %v",
+					objectsToRetry.oType, areEqual)
+				// QUESTION: should I do anything at all with the retry entries if the two objects are EQUAL?
+				// maybe set ignore=true?
+				if areEqual {
+					return
```

Comment:
maybe a debug level log here would be good

Reply:
I already have this line just above that tells us whether they two objects are considered equal:
```
klog.V(5).Infof("Update event received for resource %s, old object is equal to new: %v",
					objectsToRetry.oType, areEqual)
```
So either I keep it as it is now, or I remove that line and add a debug print under the if clause above (`if areEqual`...)

Reply:
sorry missed that, all good.